### PR TITLE
storageclass: add glog to Delete(); return improvements

### PR DESCRIPTION
**Pull request by sebrandon1**
Builds on changes from: #379 

As noted in #672 we should have a `glog` entry for when an object does not exist.

### CODE REVIEW

File: pkg/storage/storageclass.go [hunk 1]
```diff
@@ -370,13 +370,13 @@ func (builder *ClassBuilder) validate() (bool, error) {
...
 
-		builder.errorMsg = msg.UndefinedCrdObjectErrString(resourceCRD)
+		return false, fmt.Errorf(msg.UndefinedCrdObjectErrString(resourceCRD))
 	}
 
 	if builder.apiClient == nil {
 		glog.V(100).Infof("The %s builder apiclient is nil", resourceCRD)
 
-		builder.errorMsg = fmt.Sprintf("%s builder cannot have nil apiClient", resourceCRD)
+		return false, fmt.Errorf("%s builder cannot have nil apiClient", resourceCRD)
```
Comment (klaskosk) [inline pkg/storage/storageclass.go:379]: resourceCRD should be updated to `storageClass` so that the error starts with a lowercase

Comment (sebrandon1) [reply pkg/storage/storageclass.go:379]: Done.

**Review by klaskosk — APPROVED** [review]
/lgtm

**Comment by klaskosk** [comment]
approved too soon again lol, but just change to lowercase in the other areas and it's fine

**Review by klaskosk — APPROVED** [review]

File: pkg/storage/storageclass.go [hunk 2]
```diff
@@ -79,9 +83,7 @@ func (builder *ClassBuilder) WithReclaimPolicy(
 		glog.V(100).Infof("The reclaimPolicy of the storageclass is empty")
 
 		builder.errorMsg = "storageclass 'reclaimPolicy' cannot be empty"
-	}
 
-	if builder.errorMsg != "" {
```
Comment (kononovn) [inline pkg/storage/storageclass.go:84]: If you remove this condition we add invalid reclaimPolicy to builder. Please revert it back

File: pkg/storage/storageclass.go [hunk 3]
```diff
@@ -101,9 +103,7 @@ func (builder *ClassBuilder) WithVolumeBindingMode(
 		glog.V(100).Infof("The bindingMode of the storageclass is empty")
 
 		builder.errorMsg = "storageclass 'bindingMode' cannot be empty"
-	}
 
-	if builder.errorMsg != "" {
```
Comment (kononovn) [inline pkg/storage/storageclass.go:106]: If you remove this condition we add invalid bindingMode to builder. Please revert it back

**Review by kononovn — CHANGES_REQUESTED** [review]
Please check comments

File: pkg/storage/storageclass.go (hunk 2 above)
Comment (sebrandon1) [reply pkg/storage/storageclass.go:84]: The diff here on Github makes it look like that would happen.  It makes it look confusing when its not.

image

File: pkg/storage/storageclass.go (hunk 3 above)
Comment (sebrandon1) [reply pkg/storage/storageclass.go:106]: Same here, the diff on Github makes it look confusing for some reason.  The actual code won't allow an invalid bindingMode though.

image

File: pkg/storage/storageclass.go [hunk 4]
```diff
@@ -246,6 +246,10 @@ func (builder *ClassBuilder) Delete() error {
 	glog.V(100).Infof("Deleting storageclass %s", builder.Definition.Name)
 
 	if !builder.Exists() {
+		glog.V(100).Infof("storageclass %s does not exist", builder.Definition.Name)
+
+		builder = nil
```
Comment (kononovn) [inline pkg/storage/storageclass.go:251]: Please fix it. builder.Object = nil

**Review by kononovn — CHANGES_REQUESTED** [review]
1 comment to fix

File: pkg/storage/storageclass.go (hunk 4 above)
Comment (sebrandon1) [reply pkg/storage/storageclass.go]: Done!

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**